server/tests/src/test/java/org/infinispan/server/persistence/JGroupsJdbcPing2IT.java

```diff
@@ -230,4 +236,109 @@ public Logger getParentLogger() {
+       * Attempts to locate the {@link Driver} that accepts the given URL.
+       *
+       * <p>
+       * This method accept custom {@link ClassLoader} to load the {@link Driver}. First, it will delegate the call to
+       * {@link DriverManager#getDriver(String)}, and if it fails, it will proceed to load the driver through the
+       * {@link ServiceLoader#load(Class, ClassLoader)} interface.
+       * </p>
+       *
+       * @param url a database URL of the form
+       *      *     <code>jdbc:<em>subprotocol</em>:<em>subname</em></code>
+       * @param cls the class loaders to load the {@link Driver} instance.
+       * @return A {@link Driver} capable of understanding the given URL.
+       * @throws SQLException If a {@link Driver} is not found.
+       */
+      public static Driver getDriver(String url, ClassLoader[] cls) throws SQLException {
+         try {
+            return DriverManager.getDriver(url);
+         } catch (SQLException ignore) { }
+
+         for (ClassLoader loader : cls) {
+            Optional<ServiceLoader.Provider<Driver>> o = ServiceLoader.load(java.sql.Driver.class, loader).stream()
+                  .filter(d -> acceptUrl(d.get(), url))
+                  .findFirst();
+
+            if (o.isEmpty()) continue;
+
+            Driver driver = o.map(ServiceLoader.Provider::get).orElseThrow();
+            // Register the driver again so subsequent invocations should succeed without using the ServiceLoader.
+            DriverManager.registerDriver(new Driver() {
+               @Override
+               public Connection connect(String url, Properties info) throws SQLException {
+                  return driver.connect(url, info);
+               }
+
+               @Override
+               public boolean acceptsURL(String url) throws SQLException {
+                  return driver.acceptsURL(url);
+               }
+
+               @Override
+               public DriverPropertyInfo[] getPropertyInfo(String url, Properties info) throws SQLException {
+                  return driver.getPropertyInfo(url, info);
+               }
+
+               @Override
+               public int getMajorVersion() {
+                  return driver.getMajorVersion();
+               }
+
+               @Override
+               public int getMinorVersion() {
+                  return driver.getMinorVersion();
+               }
+
+               @Override
+               public boolean jdbcCompliant() {
+                  return driver.jdbcCompliant();
+               }
+
+               @Override
+               public Logger getParentLogger() throws SQLFeatureNotSupportedException {
+                  return driver.getParentLogger();
+               }
+            });
+            return driver;
+         }
+
+         throw new SQLException("No suitable driver", "08001");
+      }
+
+      public static Connection getConnection(String url, String user, String password, ClassLoader[] cls) throws SQLException {
```

Comment:
Should we instead call `DriverManager.getConnection(connectionUrl, username, password);` here like it was before and then do the catch?